### PR TITLE
Bugfix: Updating Recipient Business Categories

### PR DESCRIPTION
**Pull request by dpb-bah**
**High level description:**
"Lockheed Martin" was being listed as a "small business". Looking into the data, there was a logic error in the Business Categories functions which resulted in inaccurate data. When fixing this, we've also updated the business categories logic based on the mappings file attached to the ticket.

**Technical details:**
* Updated the business categories functions with the logic below reflected in the new mappings file attached to the ticket.
	* Business - "category_business"
	    - Adding
		    Tax Exempt - "corporate_entity_tax_exempt"
		    Not Tax Exempt - "corporate_entity_not_tax_exempt"
		    Partnership - "partnership_or_limited_liability_partnership"
		    Sole Proprietorship - "sole_proprietorship"
		    Manufacturer of Goods - "manufacturer_of_goods"
		    Sub-Chapter S Corporation - "subchapter_s_corporation"
		    Limited Liability Corporation (LLC) - "limited_liability_corporation"
	    - Modified 
		Small Business - "small_business"
                 Other than Small Business - "other_than_small_business"
	* Special Business Designations - "special_designations"
	    - Adding
		Domestic Shelter - "domestic_shelter"
		Hospital - "hospital"
		Veterinary Hospital - "veterinary_hospital"
	* Higher Education - "higher_education"
	    - Adding
		Educational Institution - "educational_institution"
		School of Forestry - "school_of_forestry"
		Veterinary College - "veterinary_college"
	* Government - "government"
	    - Making it's own subcategory from previous subcategories
		Interstate Entity - "interstate_entity"
		Regional Organization - "regional_organization"
		Council of Governments - "council_of_governments"
* Updated the resulting business categories listed in the Legal Entity model
* Updated the other files which use the business categories functions

**Link to JIRA Ticket:**
[DEV-626](https://federal-spending-transparency.atlassian.net/browse/DEV-626)

**The following are ALL required for the PR to be merged:**
- [x] Backend review completed
- [x] Frontend impact assessment completed [Recipient Types Gist](https://github.com/fedspendingtransparency/usaspending-api/wiki/Recipient-Business-Types)
- [x] Data validation completed
- [x] API Performance evaluation completed and present N/A

### CODE REVIEW

File: usaspending_api/broker/helpers.py [hunk 1]
```diff
@@ -322,20 +329,61 @@ def get_business_categories(row, data_type):
...
+        # BUSINESS (FOR-PROFIT)
+        if row.get('contracting_officers_deter') == 'S' \
+                or legal_entity_bool_dict['women_owned_small_business'] is True \
+                or legal_entity_bool_dict['economically_disadvantaged'] is True \
+                or legal_entity_bool_dict['joint_venture_women_owned'] is True \
+                or legal_entity_bool_dict['emerging_small_business'] is True \
+                or legal_entity_bool_dict['self_certified_small_disad'] is True \
+                or legal_entity_bool_dict['small_agricultural_coopera'] is True \
+                or legal_entity_bool_dict['small_disadvantaged_busine'] is True:
+            business_category_set |= {'small_business'}
```
Comment (tony-sappe) [inline usaspending_api/broker/helpers.py]: Why are the types sometimes added via `.add()` and other times using the union operator `|=`?

Comment (dpb-bah) [reply usaspending_api/broker/helpers.py]: Good call, never noticed that. I'll stick with the consistent `.add()`

File: usaspending_api/broker/management/sql/create_business_categories_functions.sql [hunk 2]
```diff
@@ -1,16 +1,162 @@
...
+)
 returns text[] as $$
 declare
     bc_arr text[];
 begin
 
-    -- SMALL BUSINESS
-    if small_business_competitive IS TRUE or for_profit_organization IS TRUE
+-- BUSINESS (FOR-PROFIT ORGANIZATION)
+    if contracting_officers_deter = 'S' or women_owned_small_business IS TRUE or economically_disadvantaged IS TRUE or joint_venture_women_owned IS TRUE or emerging_small_business IS TRUE or self_certified_small_disad IS TRUE or small_agricultural_coopera IS TRUE or small_disadvantaged_busine IS TRUE
```
Comment (tony-sappe) [inline usaspending_api/broker/management/sql/create_business_categories_functions.sql]: Can this be split into multiple lines for readability?

File: usaspending_api/broker/management/sql/create_business_categories_functions.sql [hunk 3]
```diff
@@ -1,16 +1,162 @@
...
+    then
+        bc_arr := bc_arr || array['subchapter_s_corporation'];
+    end if;
+
+    if limited_liability_corporat IS TRUE
+    then
+        bc_arr := bc_arr || array['limited_liability_corporation'];
+    end if;
+
+    if for_profit_organization IS TRUE or (bc_arr && array['small_business', 'other_than_small_business', 'corporate_entity_tax_exempt','corporate_entity_not_tax_exempt', 'partnership_or_limited_liability_partnership', 'sole_proprietorship', 'manufacturer_of_goods', 'subchapter_s_corporation', 'limited_liability_corporation'])
```
Comment (tony-sappe) [inline usaspending_api/broker/management/sql/create_business_categories_functions.sql]: Split for readability?

File: usaspending_api/broker/management/sql/create_business_categories_functions.sql (hunk 2 above)
Comment (dpb-bah) [reply usaspending_api/broker/management/sql/create_business_categories_functions.sql]: Fixed in latest commit.

File: usaspending_api/broker/management/sql/create_business_categories_functions.sql (hunk 3 above)
Comment (dpb-bah) [reply usaspending_api/broker/management/sql/create_business_categories_functions.sql]: Fixed in latest commit.

File: usaspending_api/broker/helpers.py [hunk 4]
```diff
@@ -280,6 +280,7 @@ def get_business_categories(row, data_type):
     business_category_set = set()
 
     if data_type == 'fabs':
```
Comment (tony-sappe) [inline usaspending_api/broker/helpers.py]: Verified that this `fabs` code matches excel doc attached to Jira Ticket

**Comment by dpb-bah** [comment]
Note: the following columns have not been added as the broker currently doesn't support them. They will be included in the future.
```
SBA Certified Samll Disadvantaged Business
SBA Certified 8A Program Participant
SBA Certified HUB Zone
DOT Certified Disadvantaged Business Enterprise
Sheltered Workshop
```

File: usaspending_api/broker/helpers.py [hunk 5]
```diff
@@ -540,7 +612,8 @@ def get_business_categories(row, data_type):
 
         if business_category_set & {'national_government', 'regional_and_state_government',
                                     'us_territory_or_possession', 'local_government',
```
Comment (tony-sappe) [inline usaspending_api/broker/helpers.py]: is `us_territory_or_possession` applicable to fpds?

**Pull request merged**